src/number_theory/pell.lean

```diff
@@ -144,6 +144,125 @@ lemma x_neg (a : solution₁ d) : (-a).x = -a.x := rfl
 @[simp]
 lemma y_neg (a : solution₁ d) : (-a).y = -a.y := rfl
 
+/-- A solution has `x ≠ 0`. -/
+lemma x_ne_zero (h₀ : 0 < d) (a : solution₁ d) : a.x ≠ 0 :=
```

Comment:
```suggestion
lemma x_ne_zero (h₀ : 0 ≤ d) (a : solution₁ d) : a.x ≠ 0 :=
```

Reply:
I can do that, but (i) Pell's equation for `d = 0` is not really interesting,and (ii) it will cause an overlong line later (line 257) beacuse of the need of adding `.le` in the invocation of the lemma.

Reply:
I don't think either of those are particularly good arguments

Reply:
OK, let me paraphrase: This would be about the only lemma with a slightly weaker hypothesis on `d`. So the question is whether we gain more by stating it in this generality than we lose by having to turn the strict inequality hypothesis that we usually have available into the non-strict one whenever we want to use it. I'm not sure that we gain more than we lose, in particular since I cannot really think of an opportunity when I would want to use `x_ne_zero` when `d = 0`. But I'm open to being convinced otherwise.

Reply:
I think you're overstating the cost of writing the three characters `.le`. Even if the result isn't useful in its full generality, it's generally a good idea to state things that way, especially when the added difficulty to the caller is only three characters.

Reply:
It's not the three characters, but the fact that you cannot directly use the hypothesis that you have, so you need to figure out how to modify it, and this creates friction.

Reply:
OK, I give in :smile: